rfcs/20190610-standartizing-composite_ops.md

```diff
@@ -0,0 +1,349 @@
+# Standardizing composite ops in tensorflow.
+
+Status        | Proposed
+:------------ | :------------------------------------
+**Author(s)** | Mark Sandler (sandler@google.com)
+**Sponsor**   | Alexandre Passos (apassos@google.com)
+**Updated**   | 2019-06-10
+
+## Objective
+
+The objective is to create a simple API that enables adding new composite ops in
+a way that they can be automatically and robustly processed by downstream
+tooling. The developers should be able to iterate on their implementation and
+potentially add them as standard tf_op, without breaking existing tools (which
+might not even be aware of their existence).
+
+Why? Composite ops often provide building blocks for building complex models.
+However, and this is especially true for embedded and specialized hardware,
+these ops when implemented naively, become unusable due to architectural
+(hardware) details. It is thus preferable for the downstream tools to be able to
+extract such composite ops and for instance provide specialized implementation.
+
+### Goals:
+
+*   Create a standard way for tensorflow community to implement re-usable ops
```

Comment:
How is this related to @qlzh727's runtime function grappler rewrite work?  can you add a section on that.  does it supercede it?

Reply:
I don't believe so. Grappler can use this information in any way it wants including ignoring it.